src/watertap_contrib/seto/multiperiod/VAGMD_batch_flowsheet_multiperiod.py

```diff
@@ -0,0 +1,193 @@
+        # initialization_options=None,
+        # unfix_dof_options=None,
+    )
+
+    active_blks = mp.get_active_process_blocks()
+
+    # Set-up for the first time period
+    active_blks[0].fs.vagmd.feed_props[0].conc_mass_phase_comp["Liq", "TDS"].fix(
+        feed_salinity
+    )
+    active_blks[0].fs.vagmd.feed_props[0].temperature.fix(feed_temp + 273.15)
+    active_blks[0].fs.acc_distillate_volume.fix(0)
+    active_blks[0].fs.pre_feed_temperature.fix(feed_temp + 273.15)
+    active_blks[0].fs.pre_permeate_flow_rate.fix(0)
+    active_blks[0].fs.acc_thermal_energy.fix(0)
+
+    return mp
+
+
+#%% Local test
+if __name__ == "__main__":
+    mp = create_multiperiod_vagmd_batch_model(
+        n_time_points=72,
+        feed_flow_rate=600,
+        evap_inlet_temp=80,
+        cond_inlet_temp=25,
+        feed_temp=25,
+        feed_salinity=35,
+        initial_batch_volume=50,
+        module_type="AS7C1.5L",
+        high_brine_salinity=False,
+        cooling_system_type="closed",
+    )
+    blks = mp.get_active_process_blocks()
+    print("degree of freedom: ", degrees_of_freedom(mp))
+    solver = get_solver()
+    results = solver.solve(mp)
+    print(
+        "{:<3}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}".format(
+            "t  ",
+            "t_minute",
+            "S",
+            "Pflux",
+            "AccVd",
+            "Ttank",
+            "TEO",
+            "TCO",
+            "RR",
+            "ThPower",
+            "AccThEnergy",
+            "PFR",
+        )
+    )
+    for i in range(len(blks)):
+        print(
+            "{:<3}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}{:<10}".format(
+                round(i),
+                round(value(blks[i].fs.vagmd.dt) * i / 60, 4),
+                round(
+                    value(
+                        blks[i]
+                        .fs.vagmd.feed_props[0]
+                        .conc_mass_phase_comp["Liq", "TDS"]
+                    ),
+                    4,
+                ),
+                round(value(blks[i].fs.vagmd.permeate_flux), 4),
+                round(value(blks[i].fs.acc_distillate_volume), 4),
+                round(value(blks[i].fs.vagmd.feed_props[0].temperature - 273.15), 4),
+                round(
+                    value(
+                        blks[i].fs.vagmd.evaporator_out_props[0].temperature - 273.15
+                    ),
+                    4,
+                ),
+                round(
+                    value(blks[i].fs.vagmd.condenser_out_props[0].temperature - 273.15),
+                    4,
+                ),
+                round(value(blks[i].fs.acc_recovery_ratio), 6),
+                round(value(blks[i].fs.vagmd.thermal_power), 4),
+                round(value(blks[i].fs.acc_thermal_energy), 4),
+                round(
+                    value(blks[i].fs.vagmd.permeate_props[0].flow_vol_phase["Liq"]), 8
+                ),
+            )
```

Comment:
I wonder if these items can just be added to the unit models ` _get_performance_contents()` method. Then, you could just call `blks[i].fs.vagmd.report()`, at least for the vagmd relevant outputs

Reply:
Yea, that's a good idea!  This part is for testing purpose and supposed to be removed after the I finalize this model. Will follow up with this comment later.